distribution/pull_v2.go

```diff
@@ -356,6 +360,12 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named) (tagUpdat
 		return false, fmt.Errorf("image manifest does not exist for tag or digest %q", tagOrDigest)
 	}
 
+	if m, ok := manifest.(*schema2.DeserializedManifest); ok {
+		if m.Manifest.Config.MediaType == schema2.MediaTypePluginConfig {
```

Comment:
If at this point the MediaType could still be `application/octet-stream` I would prefer to update it to `schema2.MediaTypeConfig` explicitly in the code with a comment linking to the issue, then check for `schema2.MediaTypeConfig`. @aaronlehmann would it problematic for verification if we updated this value before the check?